manim/mobject/types/vectorized_mobject.py

```diff
@@ -906,6 +906,185 @@ def __init__(self, *vmobjects, **kwargs):
+            raise Exception("All submobjects must be of type VMobject")
+        VMobject.__init__(self, **kwargs)
+        self.show_keys = show_keys
+        self.submob_dict = {}
+        self.add(*pairs)
+        
+
+    def add(self, *pairs):
+        '''Adds the key-value pairs to the :class:`VDict` object.
+
+            Also, it internally adds the value to the `submobjects` :class:`list`
+            of :class:`~.Mobject`, which is responsible for actual on-screen display
+            
+            Paramters
+            ---------
+            *pairs : :class:`iterable`
+                Each pair is a :class:`tuple` wherein the first 
+                element is the key for the mobject and the second
+                element is the actual mobject
+
+            Returns
+            -------
+            self : :class:`VDict`
+                Returns the :class:`VDict` object on which this method was called
+
+            Examples
+            --------
+            Normal usage::
+                square_obj = Square()
+                my_dict.add(('s', square_obj))
+            
+        '''
+        for pair in pairs:
+            key = pair[0]
+            value = pair[1]
+            
+            mob = value
+            if self.show_keys:
+                # This import is here and not at the top to avoid circular import
+                from ...mobject.svg.tex_mobject import TextMobject
+                key_text = TextMobject(str(key)).next_to(value, LEFT)
+                mob.add(key_text)
+            
+            self.submob_dict[key] = mob
+            super().add(value)
+        return self
+
+    def remove(self, key):
+        '''Removes the mobject from the :class:`VDict` object having the key `key`
+        
+            Also, it internally removes the mobject from the `submobjects` :class:`list`
+            of :class:`~.Mobject`, (which is responsible for removing it from the screen)
+
+            Parameters
+            ----------
+            key : 
+                The key of the submoject to be removed
+
+            Returns
+            -------
+            self : :class:`VDict`
+                Returns the :class:`VDict` object on which this method was called
+
+            Examples
+            --------
+            Normal usage::
+            my_dict.remove('square')
+
+        '''
+        if key not in self.submob_dict:
+            raise Exception("The given key '%s' is not present in the VDict" %str(key))
+        super().remove(self.submob_dict[key])
+        del self.submob_dict[key]
+        return self
+
+    def __getitem__(self, key):
+        '''Overriding the [] operator for getting submobject by key
+        
+            Parameters
+            ----------
+            key : any suitable type of a python `dict`
+                The key of the submoject to be accessed
+            
+            Returns
+            -------
+            submob : :class:`VMobject`
```

Comment:
Return values aren't named
```suggestion
            :class:`VMobject`
```